### PR TITLE
Support php-cs-fixer

### PR DESCRIPTION
**Pull request by zdenekdrahos**
Closes https://github.com/EdgedesignCZ/phpqa/issues/60

- [x] configure dry-run and rules in `.phpqa.yml`
- [x] html report - [demo](https://edgedesigncz.github.io/phpqa/report/php-cs-fixer.html)
- [x] add as suggested tool
- [x] `bin/suggested-tools.sh install` doesn't work in default "clone", the tool needs [Symfony3 components](https://github.com/EdgedesignCZ/phpqa#symfony3-components)

```bash
# install php-cs-fixer after git clone
composer require friendsofphp/php-cs-fixer symfony/filesystem symfony/config symfony/process symfony/finder symfony/console symfony/event-dispatcher phploc/phploc
```

### CODE REVIEW

**Comment by zdenekdrahos** [comment]
@keradus, @mastercoding: could you try this version? thanks

```
composer update edgedesign/phpqa:dev-php-cs-fixer
```

File: README.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Experimental tool is executed only if the tool is specified in `--tools`.
 
 Tool | PHP | Supported since | Description | Status |
 ---- | --- | --------------- | ----------- | ------ |
+[php-cs-fixer](http://cs.sensiolabs.org/) | `>= 5.6` | `1.12` | Automatically detect and fix PHP coding standards issues | stable |
```
Comment (keradus) [inline README.md]: FYI: if you want support for lower PHP, go with PHP CS Fixer v2.2+

File: README.md [hunk 2]
```diff
@@ -199,6 +201,8 @@ Tool | Settings | Default Value | Your value
 [phpcs.standard](https://pear.php.net/manual/en/package.php.php-codesniffer.usage.php#package.php.php-codesniffer.usage.coding-standard) | Coding standard | PSR2 | Name of existing standard (`PEAR`, `PHPCS`, `PSR1`, `PSR2`, `Squiz`,  `Zend`), or path to your coding standard
 [phpcs.ignoreWarnings](https://github.com/EdgedesignCZ/phpqa/issues/53) | If number of allowed errors is compared with warnings+errors, or just errors from `checkstyle.xml` | `false` | Boolean value
 [phpcs.reports](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting) | Report types | [`full`](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting#printing-full-and-summary-reports) report in [cli mode](#output-modes), [`checkstyle`](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting#printing-a-checkstyle-report) in [file mode](#output-modes) | Predefined [report types](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting) or [custom reports](https://github.com/wikidi/codesniffer#examples)
+[php-cs-fixer.rules](http://cs.sensiolabs.org/#usage) | Coding standard rules | `@PSR2` | String value
```
Comment (keradus) [inline README.md]: you may require `risky` support, as some of the rules required that to be enabled

File: app/report/php-cs-fixer.xsl [hunk 3]
```diff
@@ -0,0 +1,126 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
+
+    <xsl:output method="html"  encoding="UTF-8"/>
+    
+    <xsl:template match="/">
+        <html>
+            <head>
+                <title>php-cs-fixer report</title>
```
Comment (keradus) [inline app/report/php-cs-fixer.xsl]: `PHP CS Fixer report`
(tool name is PHP CS Fixer, not php-cs-fixer)

File: src/CodeAnalysisTasks.php [hunk 4]
```diff
@@ -384,13 +392,37 @@ function ($relativeDir) {
         );
     }
 
+    private function phpcsfixer()
```
Comment (keradus) [inline src/CodeAnalysisTasks.php]: if you like, you could execute the code (FixCommand) instead of running it as separated CLI process

**Comment by mastercoding** [comment]
Hi @zdenekdrahos! Great to see that you tackled this so soon. I would propose two configurable options in addition to what you've got:

- using-cache (yes/no)
- config

The configuration file i've got for php-cs-fixer is complex (https://gist.github.com/mastercoding/49db81e63b5b849779ecb79bd1e896e0). It would almost be impossible to put that in your tool.

What do you think/

**Comment by keradus** [comment]
yeah, let point config file instead of mapping the properties

**Comment by zdenekdrahos** [comment]
Thanks for great comments.

Cache can be disabled in [config file](http://cs.sensiolabs.org/#caching). Is it necessary to define _cache_ in `.phpqa.yml`?

**Comment by zdenekdrahos** [comment]
@keradus, @mastercoding: is current version ok? Or is there something you'd like to change?

File: .gitignore [hunk 5]
```diff
@@ -2,3 +2,4 @@
 /vendor/
 /build/
 phpstan-phpqa.neon
+.php_cs.cache
```
Comment (keradus) [inline .gitignore]: I would add `.php_cs` (not `.php_cs.dist`) as well

File: src/CodeAnalysisTasks.php [hunk 6]
```diff
@@ -384,13 +392,45 @@ function ($relativeDir) {
...
+    {
+        $analyzedDirs = $this->options->getAnalyzedDirs();
+        $analyzedDir = reset($analyzedDirs);
+        if (count($analyzedDirs) > 1) {
+            $this->say("<error>php-cs-fixer analyzes only first directory {$analyzedDir}</error>");
```
Comment (keradus) [inline src/CodeAnalysisTasks.php]: huh? you could provide multiple paths (dirs or files) in single CLI run

**Pull request merged**